### ISSUE TITLE
New Rule: no-script-url

### ISSUE THREAD

**Issue by Gaic4o** [eslint/markdown#694]
### Rule details

I would like to propose a rule that reports `javascript:` URLs used as Markdown link destinations.  A `javascript:` URL can be evaluated as JavaScript code by a browser. Some Markdown renderers reject these URLs during rendering, but the same document may be processed by different renderers with different configurations, so it would be useful to detect them at the source lint stage as well.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
[View documentation](javascript:alert(1))

[View documentation][documentation]

[documentation]: javascript:alert(1)

<javascript:alert(1)>

[Mixed case scheme](JavaScript:alert(1)) 
```

### Prior Art

ESLint core's [`no-script-url`](https://eslint.org/docs/latest/rules/no-script-url) rule reports `javascript:` URLs in string literals and statically evaluable untagged template literals. Although it is not a Markdown linter, it provides prior art for the rule name and behavior.

[`jsx-no-script-url`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-script-url.md) from `eslint-plugin-react` provides a similar check for link-related JSX attributes.

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Detecting `javascript:` links during source linting can help prevent potentially unsafe links from being added to a repository by mistake or through external contributions.

**Comment by nzakas** [comment]
Thanks for the suggestion. I don't think this is general enough to include in this plugin, but you can always make a custom rule that does this for you.
